services/time/src/lib.rs

```diff
@@ -14,10 +14,91 @@
+//! (usually immediately after the commit of each block). The time service maintains an index with the
+//! most current time values indicated separately by each validator. This index is updated after each
+//! transaction from any of the validators. A 1/3 quantile of these values (ordered by decreasing time)
+//! is then picked out from the index; this quantile is considered the actual time and is applied to
+//! determine the exact service time.
+//!
 //! See [the Exonum documentation][docs:time] for a high-level overview of the service,
 //! in particular, its design rationale and the proof of correctness.
 //!
 //! [docs:time]: https://exonum.com/doc/version/latest/advanced/time
+//!
+//! # HTTP API
+//!
+//! REST API of the service is documented in the [`api` module](api/index.html).
+//!
+//! # Examples
+//!
+//! ## Use with TestKit
+//!
+//! ```
+//! use chrono::{TimeZone, Duration, Utc};
+//! use exonum::{
+//!     helpers::Height,
+//!     runtime::{InstanceId, SnapshotExt},
+//! };
+//! use exonum_rust_runtime::ServiceFactory;
+//! use exonum_time::{MockTimeProvider, TimeProvider, TimeServiceFactory, TimeSchema};
+//! use exonum_testkit::{ApiKind, TestKit, TestKitBuilder};
+//!
+//! use std::sync::Arc;
+//!
+//! const TIME_SERVICE_ID: InstanceId = 100;
+//! const TIME_SERVICE_NAME: &'static str = "time-oracle";
+//!
+//! // Time provider for a time service is chosen at the compile time.
+//! // Here we use `MockTimeProvider` to get controllable results.
+//! // In real world it makes sense to use `SystemTimeProvider` or your own implementation
+//! // of `TimeProvider`.
+//! let time_provider = Arc::new(MockTimeProvider::default());
+//!
+//! // Factory for time service will create instances of the service with given
+//! // time provider.
+//! let time_service_factory =
+//!     TimeServiceFactory::with_provider(time_provider.clone() as Arc<dyn TimeProvider>);
+//! let time_service_artifact = time_service_factory.artifact_id();
+//!
+//! // Create testkit with the time service.
+//! let mut testkit: TestKit = TestKitBuilder::validator()
+//!     .with_artifact(time_service_artifact.clone())
+//!     .with_instance(
+//!            time_service_artifact.into_default_instance(TIME_SERVICE_ID, TIME_SERVICE_NAME),
+//!        )
+//!     .with_rust_service(time_service_factory)
+//!     // Add other services here
+//!     .create();
+//!
+//! // Set time in `MockTimeProvider`.
+//! time_provider.set_time(Utc.timestamp(10, 0));
+//!
+//! // Create some blocks for time to appear in the blockchain.
+//! testkit.create_blocks_until(Height(2));
+//!
+//! // Obtain time service schema.
+//! let snapshot = testkit.snapshot();
+//! let time_schema: TimeSchema<_> = snapshot.service_schema(TIME_SERVICE_NAME).unwrap();
+//!
+//! // Obtain time from the schema. Service can base its logic on this time.
+//! let time = time_schema.time.get();
+//!
+//! // With `MockServiceProvider` we can ensure that time is based on data provided by `TimeProvider`.
```

Comment:
Micro-nit: This line seems overly long (doesn't fit into layout on my laptop). Some other lines (e.g., `time_schema` instantiation) could be shortened as well.

Reply:
I've split comment lines that exceed 80 symbols limit into several lines, but left source lines as-is, since they don't violate 100 symbols limit (default for source code lines as per [`rustfmt` guideline](https://github.com/rust-dev-tools/fmt-rfcs/blob/master/guide/guide.md)) -- splitting them will make code in docs look unnaturally formatted.